src/ui/pages/SignTypedData/SignTypedData.tsx

```diff
@@ -337,34 +339,42 @@ function TypedDataDefaultView({
               >
                 Cancel
               </Button>
-
-              <SignMessageButton
-                wallet={wallet}
-                ref={signMsgBtnRef}
-                onClick={() => {
-                  if (!seenSigningData) {
-                    scrollSigningData();
-                  } else {
-                    signTypedData_v4();
+              {preferences ? (
+                <SignMessageButton
+                  wallet={wallet}
+                  ref={signMsgBtnRef}
+                  onClick={() => {
+                    if (!seenSigningData) {
+                      scrollSigningData();
+                    } else {
+                      signTypedData_v4();
+                    }
+                  }}
+                  buttonKind={
+                    interpretationHasCriticalWarning ? 'danger' : 'primary'
                   }
-                }}
-                buttonKind={
-                  interpretationHasCriticalWarning ? 'danger' : 'primary'
-                }
-                buttonTitle={
-                  interpretationHasCriticalWarning
-                    ? 'Proceed Anyway'
-                    : undefined
-                }
-                children={
-                  seenSigningData ? null : (
-                    <HStack gap={8} alignItems="center" justifyContent="center">
-                      <span>Scroll</span>
-                      <ArrowDownIcon style={{ width: 24, height: 24 }} />
-                    </HStack>
-                  )
-                }
-              />
+                  buttonTitle={
+                    interpretationHasCriticalWarning
+                      ? 'Proceed Anyway'
+                      : undefined
+                  }
+                  children={
+                    seenSigningData ? null : (
+                      <HStack
+                        gap={8}
+                        alignItems="center"
+                        justifyContent="center"
+                      >
+                        <span>Scroll</span>
+                        <ArrowDownIcon style={{ width: 24, height: 24 }} />
+                      </HStack>
+                    )
+                  }
+                  holdToSign={
+                    preferences.enableHoldToSignButton && seenSigningData
```

Comment:
do we need to check `seenSigningData` here? I think we're not rendering children anyway if it's `false`

Reply:
I need to double check, but this looks like a fair condition anyway. If we show `Scroll` button, we don't need to use holdable state

Reply:
It looks like we are rendering them when it is false, btw. Anyway, let me check :)

Reply:
Yes, this condition is important